### PR TITLE
-Wno-error=comment is needed when compiling HLS with g++

### PR DESCRIPTION
**Pull request by sjalander**
The MLIR headers that the HLS backend depends on have comments with three slashes '///', which cause g++ to throw an error unless -Wno-error=comment is used.

This case is triggered if jlm is configured to use g++, i.e., ./configure.sh --circt-path ../circt/local

### CODE REVIEW

File: configure.sh [hunk 1]
```diff
@@ -85,7 +85,7 @@ else
 fi
 
 if [ "${CIRCT_ENABLED}" == "yes" ] ; then
-	CPPFLAGS_CIRCT="-DCIRCT=1 -I${CIRCT_PATH}/include"
```
Comment (phate) [inline configure.sh:88]: If I am not mistaken, then the `-Wno-error=comment` should go into `CXXFLAGS_CIRCT` and not `CPPFLAGS_CIRCT`.

**Review by phate — CHANGES_REQUESTED** [review]

File: configure.sh (hunk 1 above)
Comment (phate) [reply configure.sh:88]: Also, why was this not caught by the CI? Is it because we only compile HLS backend with clang?

Comment (sjalander) [reply configure.sh:88]: That's the case.

Comment (sjalander) [reply configure.sh:88]: Changed to CXXFLAGS_CIRCT

**Review by phate — APPROVED** [review]

**Pull request merged**